### PR TITLE
Fix find xml from l1c

### PR DESCRIPTION
**Pull request by ninahakansson**
For PPSv2021 when we have the level1c file, all other files produced by PPS can be derived from it. Use the level1c file to find the xml files.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed: Passes ``pytest nwcsafpps_runner`` 
 - [x] Passes ``git diff origin/main -- "*py" | flake8 --diff main`` 
 - [ ] Fully documented

### CODE REVIEW

**Comment by ninahakansson** [comment]
When I click on details for the failed coveralls -run I see different numbers: coverage increased (+0.7%) to 64.351%. I don't understand what it is unhappy about?

File: nwcsafpps_runner/tests/test_utils.py [hunk 1]
```diff
@@ -23,14 +23,66 @@
...
+    assert set(res) == set(expected)
+
+    scene = {}
+    #: Test xml files for timectrl no file4pps attribute
+    res = create_xml_timestat_from_lvl1c(scene, mydir_out)
+    assert res == []
+
+    #: Test xml files for products no file4pps attribute
+    res = create_product_statistics_from_lvl1c(scene, mydir_out)
+    assert res == []
```
Comment (mraspaud) [inline nwcsafpps_runner/tests/test_utils.py:83]: These look like 4 different tests, could they be split into different functions? Maybe into a class, or with a new fixture for the file creations?

File: nwcsafpps_runner/utils.py [hunk 2]
```diff
@@ -587,6 +587,32 @@ def get_xml_outputfiles(path, platform_name, orb, st_time=''):
...
+        return []
+
+
+def create_pps_file_from_lvl1c(l1c_file_name, pps_control_path, name_tag, file_type):
+    """From lvl1c file create name_tag-file of type file_type."""
+    l1c_path, l1c_file = os.path.split(l1c_file_name)
+    l1c_file_as_list = l1c_file.split("_")
+    l1c_file_as_list[2] = name_tag
+    l1c_file_as_list[-1] = l1c_file_as_list[-1].replace(".nc", file_type)
+    return os.path.join(pps_control_path, ("_").join(l1c_file_as_list))
```
Comment (mraspaud) [inline nwcsafpps_runner/utils.py:613]: Would it make sense to use trollsift here?

File: bin/pps2018_runner.py [hunk 3]
```diff
@@ -180,11 +182,18 @@ def pps_worker(scene, publish_q, input_msg, options):
             err_reader2.join()
 
         pps_control_path = my_env.get('STATISTICS_DIR', options.get('pps_statistics_dir', './'))
-        xml_files = create_xml_timestat_from_ascii(scene, pps_control_path)
-        xml_files = xml_files + get_product_statistics_files(pps_control_path,
-                                                             scene,
-                                                             options['product_statistics_filename'],
-                                                             options['pps_filetime_search_minutes'])
+        if "file4pps" in scene:
```
Comment (mraspaud) [inline bin/pps2018_runner.py:185]: This `if` statement is repeated in `create_xml_timestat_from_lvl1c` and `create_product_statistics_from_lvl1c`, could these function take care of calling the right function instead?

**Review by mraspaud — COMMENTED** [review]
Thanks for fixing this! just a couple of comments

File: nwcsafpps_runner/tests/test_utils.py (hunk 1 above)
Comment (ninahakansson) [reply nwcsafpps_runner/tests/test_utils.py:83]: I followed the strategy of test_outputfiles and kept them in one function to just create files ones. I read here https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html. I did not understand a way to use the tmp_path with the unittest.TestCase and setUp/tearDown. Do you know of an example where tmp_path is used in better way?

File: bin/pps2018_runner.py (hunk 3 above)
Comment (ninahakansson) [reply bin/pps2018_runner.py:185]: I was thinking that the old functions: create_xml_timestat_from_ascii and get_product_statistics_files can "soon" be removed. And for that I wanted totally separate functions in the utils.py. It is the same statement but with different meanings. Here it checks if we have version 2018 or version 2021 and in the utils.py the check handles bad input. Currently the check in utils.py is not needed as we have the check here. But when we remove the check here the functions in utils.py should have the check.

File: nwcsafpps_runner/utils.py (hunk 2 above)
Comment (ninahakansson) [reply nwcsafpps_runner/utils.py:613]: I will have a look :)
!

File: nwcsafpps_runner/tests/test_utils.py (hunk 1 above)
Comment (mraspaud) [reply nwcsafpps_runner/tests/test_utils.py:83]: Here is an example with a fixture:
https://github.com/pytroll/satpy/blob/324bf13dd63174db8bc2efeb0364660807b9cd23/satpy/tests/reader_tests/test_oceancolorcci_l3_nc.py#L93-L125

File: bin/pps2018_runner.py (hunk 3 above)
Comment (mraspaud) [reply bin/pps2018_runner.py:185]: Ok, I understand. I'm ok with leaving the logic as it is now. However, would it be possible to change that `if` statement with something more explicit? For example here, could the version of  acpg be checked directly? and in the called function, could we maybe have a try/except instead to make it clear it happens when something goes wrong?

**Comment by adybbroe** [comment]
> When I click on details for the failed coveralls -run I see different numbers: coverage increased (+0.7%) to 64.351%. I don't understand what it is unhappy about?

Don't know either, I have had trouble with understanding the Coveralls results in other projects recently. I am happy with CodeFactor that also checks test coverage:

https://app.codecov.io/gh/pytroll/pytroll-pps-runner/pull/54

**Review by adybbroe — APPROVED** [review]
LGTM. I have no further comments - I see @mraspaud has already provided good input.

File: nwcsafpps_runner/tests/test_utils.py (hunk 1 above)
Comment (mraspaud) [reply nwcsafpps_runner/tests/test_utils.py:83]: and here an example with `setup_method` https://github.com/pytroll/satpy/blob/37e77dd2921c882dd501c68fc2837942dbe51fe6/satpy/tests/reader_tests/test_ghrsst_l2.py#L35-L73

**Review by mraspaud — APPROVED** [review]
LGTM!

**Pull request merged**